src/shared/signalProcessHandlers.ts

```diff
@@ -3,28 +3,34 @@ const errorTypes = ['unhandledRejection', 'uncaughtException']
 const signalTraps = ['SIGTERM', 'SIGINT', 'SIGUSR2']
 import producer from '@kafka/producer'
 import consumer from '@kafka/consumer'
+import logger from './Logger'
 
-errorTypes.map(type => {
-  process.on(type, async () => {
-    try {
-      await producer.disconnect()
-      await consumer.disconnect()
-      console.log(`process.on ${type}`)
-      await 
-      process.exit(0)
-    } catch (_) {
-      process.exit(1)
-    }
+for (const errorType of errorTypes) {
+  process.on(errorType, () => {
+      Promise.all([
+        producer.shutdown(),
+        consumer.shutdown()
+      ]).then(() => { 
+        logger.info('All promises resolved, exiting process');
+        process.exit(0)
+      })
+      .catch(() => {
```

Comment:
Any reason to have this `.catch` in a different line?

Reply:
No, but I need to keep the consistency so I'll move it above